plugins/rikki/heroeslounge/models/Match.php

```diff
@@ -82,6 +83,16 @@ class Match extends Model
         ]
     ];
 
+    public function listWinnerOptions($fieldname, $value, $formData)
+    {
+        return [$formData["teams"][0]['id'] => $formData["teams"][0]['title'], $formData["teams"][1]['id'] => $formData["teams"][1]['title']];
+    }
+
+    public function divisionsLongTitleDropdown()
+    {
+        return Division::listDivisionsWithLongTitle();
+    }
```

Comment:
If at all possible, this should not be part of the model, but of the controller - This has nothing to do with the match at all!

Reply:
According to the [OctoberCMS documentation](https://octobercms.com/docs/backend/forms#form-fields) for form fields, the field definitions should be defined on the model class file, unless I'm misunderstanding their documentation.

For this specific case, because it's a form for creating a match, I'm quite certain that the accompanying `fields_on_create.YAML` file can only reference the match class to populate it's data.

Reply:
Could you point me to the part of the docs that make you think so? Unless my understanding of MVC is entirely off this is not something that goes in the model.

Also, it should be static if October is really so inflexible that it HAS to be there

Reply:
Moved to the match controller